unstructured/ingest/v2/processes/connectors/azure_cognitive_search.py

```diff
@@ -0,0 +1,225 @@
+import json
+import multiprocessing as mp
+import typing as t
+import uuid
+from dataclasses import dataclass, field
+from pathlib import Path
+
+from unstructured.ingest.enhanced_dataclass import enhanced_field
+from unstructured.ingest.error import DestinationConnectionError, WriteError
+from unstructured.ingest.utils.data_prep import chunk_generator
+from unstructured.ingest.utils.string_and_date_utils import ensure_isoformat_datetime
+from unstructured.ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    UploadContent,
+    Uploader,
+    UploaderConfig,
+    UploadStager,
+    UploadStagerConfig,
+)
+from unstructured.ingest.v2.logger import logger
+from unstructured.ingest.v2.processes.connector_registry import (
+    DestinationRegistryEntry,
+    add_destination_entry,
+)
+from unstructured.utils import requires_dependencies
+
+if t.TYPE_CHECKING:
+    from azure.search.documents import SearchClient
+
+
+CONNECTOR_TYPE = "azure_cognitive_search"
+
+
+@dataclass
+class AzureCognitiveSearchAccessConfig(AccessConfig):
+    key: t.Optional[str] = enhanced_field(default=None, overload_name="azure_cognitive_search_key")
+
+
+@dataclass
+class AzureCognitiveSearchConnectionConfig(ConnectionConfig):
+    endpoint: str
+    index: str
+    access_config: AzureCognitiveSearchAccessConfig = enhanced_field(sensitive=True)
+
+    @requires_dependencies(["azure.search"], extras="azure-cognitive-search")
```

Comment:
Doesn't this need both `azure.search` and `azure.core` given the dependencies being used?

Reply:
Agreed, now added with: [add missing dependency check](https://github.com/Unstructured-IO/unstructured/pull/3311/commits/c8eab930f4bcc22cdbd47632b5a7726e7cb9afd6)